script.js

```diff
@@ -0,0 +1,328 @@
+main.append(textarea);
+textarea.id = ('textarea');
+textarea.autofocus = true;
+
+const keyboard = document.createElement('div');
+keyboard.id = ('keyboard');
+main.append(keyboard);
+
+localStorage.setItem('first', '0');
+
+
+// collect the virtual keyboard and add the third element to the array id array for each element
+function keyboardBuilder(allArray) {
+  for (let i = 0; i < allArray.length; i += 1) {
+    const div = document.createElement('section');
+    keyboard.append(div);
+    div.id = array[i][2];
+    document.getElementById(div.id).innerHTML = array[i][localStorage.getItem('second')
+    || localStorage.getItem('first')];
+  }
+}
+
+
+//  highlight keys and value outputcd
+document.addEventListener('keydown', (event) => {
+  for (let i = 0; i < array.length; i += 1) {
+    if (event.code === array[i][2]) {
+      document.querySelector(`#${array[i][2]}`).classList.add('change');
+      if (i !== 60 && i !== 61 && i !== 62 && i !== 53) {
+        event.preventDefault();
+      }
+      if (i !== 13 && i !== 14 && i !== 27 && i !== 28 && i !== 41 && i !== 42 && i !== 53 && i !== 54
+         && i !== 55 && i !== 56 && i !== 57 && i !== 58 && i !== 59 && i !== 60 && i !== 61 && i !== 62 && i !== 63) {
+        const output = document.getElementById('textarea');
+        output.value += document.getElementById(array[i][2]).textContent;
+      }
+    }
+  }
+  textarea.focus();
+});
+document.addEventListener('keyup', (event) => {
+  for (let i = 0; i < array.length; i += 1) {
+    if (event.code === array[i][2]) {
+      document.querySelector(`#${array[i][2]}`).classList.remove('change');
+    }
+  }
+});
+
+// displaying values from the virtual keyboard when you click the mouse
+document.addEventListener('mousedown', (event) => {
+  for (let i = 0; i < array.length; i += 1) {
+    if (event.toElement.id === array[i][2]) {
+      if (i !== 13 && i !== 14 && i !== 27 && i !== 28 && i !== 41 && i !== 42 && i !== 53 && i !== 54
+        && i !== 55 && i !== 56 && i !== 57 && i !== 58 && i !== 59 && i !== 60 && i !== 61 && i !== 62 && i !== 63) {
+        const output = document.getElementById('textarea');
+        output.value += document.getElementById(array[i][2]).innerHTML;
+      }
+    }
+  }
+});
+//  values output from the virtual keyboard by clicking with the mouse
+document.addEventListener('mousedown', (event) => {
+  for (let i = 0; i < array.length; i += 1) {
+    if (event.toElement.id === array[i][2]) {
+      document.querySelector(`#${array[i][2]}`).classList.add('change');
+      setTimeout(() => {
+        document.querySelector(`#${array[i][2]}`).classList.remove('change');
+      }, 300);
+    }
+  }
+});
+
+// change keyboard layout
+const map = {};
+
+document.addEventListener('keydown', (event) => {
+  map[event.code] = true;
+  const flag = document.getElementById(array[15][2]);
+  if (map.ControlLeft === true && map.AltLeft === true && flag.textContent === 'q') {
```

Comment:
isn't it enough to have (map.ControlLeft && map.AltLeft && flag.textContent === 'q')